### DISCUSSION TITLE
Add jitter to backoff attempts

### DISCUSSION THREAD

**Discussion by derekperkins** [TanStack/query#4490] [Ideas]
Adding jitter to backoff attempts by default would be a significant win for backends around the world :)
https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/

**Comment by TkDodo** [comment]
interesting read, thanks for sharing! Isn't this only relevant when N clients do requests at the very same time though?

**Reply by derekperkins** [reply]
We have a number of places where there are dozens of queries firing off for dashboards and whatnot that would have less contention if they jittered on retries. It's not as obvious a win for the frontend compared to the article, but could still be helpful

**Reply by TkDodo** [reply]
I see. Do you have a custom `retry` implementation already that uses jitter? I wouldn't mind adding this to the next major versions as the default behaviour...

**Reply by derekperkins** [reply]
Not in js, it's like @tannerlinsley is a busy guy or something :) A quick search returned a number of implementations that match my API expectations, like https://github.com/lifeomic/attempt
